scripts/develop-module.js

```diff
@@ -0,0 +1,299 @@
+
+class DevelopModuleError extends Error {
+    constructor(message) {
+        super(message);
+    }
+}
+
+const args = arg({
+    '--module-name': String,
+    '--help': Boolean
+});
+const moduleName = args['--module-name'];
+const shouldHelp = args['--help'];
+
+function run() {
+    try {
+        validateArguments();
+        processRequest();
+    } catch (error) {
+        if (!(error instanceof DevelopModuleError)) {
+            throw error;
+        }
+
+        log.error(error.message);
+    }
+}
+
+function validateArguments() {
+    if (shouldHelp || moduleName) {
+        return;
+    }
+
+    throw new DevelopModuleError('Incorrect usage! Pass --help flag for help.');
+}
+
+function processRequest() {
+    if (shouldHelp) {
+        processHelpRequest();
+    } else if (moduleName) {
+        processNewModuleRequest();
+    } else {
+        throw new DevelopModuleError('Script Error! Unknown flag that passed validation to process request.');
+    }
+}
+
+function processHelpRequest() {
+    log.info(`
+USAGE:
+    --module-name
+                                The new module that you want to generate the boilerplate code for.
+                                Eg: node scripts/develop-module.js --module-name=toast-message
+    --help
+                                Used to generate this help message
+                                Eg: node scripts/develop-module.js --help
+    `);
+}
+
+function processNewModuleRequest() {
+    const newModuleData = {
+        id: moduleName.replace(/\s/g, '-').toLowerCase(),
+        moduleName
+    };
+
+    addStyles(newModuleData);
+    addDocs(newModuleData);
+    addTests(newModuleData);
+
+    log.info('Generated boilerplate code for the new module %s', moduleName);
+    log.info('Add documentation in file ~ %s', newModuleData.moduleDocFilePath);
+    log.info('Add styles in folder ~ %s', newModuleData.srcFolderPath);
+    log.info('Add tests in file ~ %s', newModuleData.testFilePath);
+}
+
+function addStyles(newModuleData) {
+    addSourceStyles(newModuleData);
+    addSourceBundleStyles(newModuleData);
+    addPackagingStyles(newModuleData);
+}
+
+function addDocs(newModuleData) {
+    addDocsModuleDoc(newModuleData);
+    addDocsModuleList(newModuleData);
+    addDocsMain(newModuleData);
+    addDocsIndex(newModuleData);
+}
+
+function addSourceStyles(newModuleData) {
+    const srcFolderPath = path.join(__dirname, '..', 'src', 'less', newModuleData.id);
+    const baseFolderPath = path.join(__dirname, '..', 'src', 'less', newModuleData.id, 'base');
+    const ds4FolderPath = path.join(__dirname, '..', 'src', 'less', newModuleData.id, 'ds4');
+    const ds6FolderPath = path.join(__dirname, '..', 'src', 'less', newModuleData.id, 'ds6');
+    newModuleData.srcFolderPath = srcFolderPath;
+
+    if (fs.existsSync(srcFolderPath)) {
+        log.warn('[STYLES][directory] Styles directory already exists!');
+        return;
+    }
+
+    fs.mkdirSync(srcFolderPath);
```

Comment:
Just pointing out, we used `fs.promises` version of this and not the sync version since it is blocking. I don't mind using the sync but this is inconsistent with the other scripts we have.

Reply:
This is a development script and a blocking call should be fine as there is nothing that the user will be doing with this node instance dedicated to the script. Dropping this comment as you are also fine keeping at `sync`